lib/matplotlib/__init__.py

```diff
@@ -761,30 +763,85 @@ def __getitem__(self, key):
 
         return self._get(key)
 
+    def get_default(self, key):
+        """Return default value for the key set during initialization."""
+        if key in _deprecated_map:
+            version, alt_key, alt_val, inverse_alt = _deprecated_map[key]
+            _api.warn_deprecated(
+                version, name=key, obj_type="rcparam", alternative=alt_key)
+            return inverse_alt(self._get(alt_key))
+
+        elif key in _deprecated_ignore_map:
+            version, alt_key = _deprecated_ignore_map[key]
+            _api.warn_deprecated(
+                version, name=key, obj_type="rcparam", alternative=alt_key)
+            return self._defaults[alt_key] if alt_key else None
+
+        return self._defaults[key]
+
+    def get_defaults(self):
+        """Return default values set during initialization."""
+        return self._defaults.copy()
+
     def _get_backend_or_none(self):
         """Get the requested backend, if any, without triggering resolution."""
         backend = self._get("backend")
         return None if backend is rcsetup._auto_backend_sentinel else backend
 
+    def __delitem__(self, key):
+        if key not in self.validate:
+            raise KeyError(
+                f"{key} is not a valid rc parameter (see rcParams.keys() for "
+                f"a list of valid parameters)")
+        try:
+            del self._rcvalues[key]
+        except KeyError as err:
+            raise KeyError(
+                f"No custom value set for {key}. Cannot delete default value."
+            ) from err
+
+    def __contains__(self, key):
+        return key in self._rcvalues
+
+    def __iter__(self):
+        """Yield from sorted list of keys"""
+        with _api.suppress_matplotlib_deprecation_warning():
```

Comment:
https://github.com/matplotlib/matplotlib/pull/12658 <- this is the ultimate source of the supression which on further discussion seems it did not do what it intended so we can probably remove this line.